torch/nn/modules/rnn.py

```diff
@@ -535,12 +537,12 @@ def check_forward_args(self, input: Tensor, hidden: Tuple[Tensor, Tensor], batch
         self.check_hidden_size(hidden[1], expected_hidden_size,
                                'Expected hidden[1] size {}, got {}')
 
-    def permute_hidden(self, hx: Tuple[Tensor, Tensor], permutation: Optional[Tensor]) -> Tuple[Tensor, Tensor]:
+    def permute_hidden(self, hx: Tuple[Tensor, Tensor], permutation: Optional[Tensor]) -> Tuple[Tensor, Tensor]:  # type: ignore
```

Comment:
Why the ignore here?

Reply:
```
torch/nn/modules/rnn.py:540: error: Argument 1 of "permute_hidden" is incompatible with supertype "RNNBase"; supertype defines the argument type as "Tensor"  [override]
torch/nn/modules/rnn.py:540: note: This violates the Liskov substitution principle
torch/nn/modules/rnn.py:540: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

Reply:
This and the `@overload` on `forward` is the same issue as for `Module.forward`. That was quite tricky to resolve, but I believe the current solution works so far (see `_forward_unimplemented` and the comment above it in `nn/modules/module.py`).

I think that's best left to a separate PR, but it would be good to link to that solution, to save the next person working on this a lot of trouble. Can you add a comment here, and refer to it from the `@overload`'s futher down @guilhermeleobas?